src/agent/rustjail/src/cgroups/fs/mod.rs

```diff
@@ -27,1120 +43,848 @@ macro_rules! sl {
-                return Ok(());
+            let kernel = memory.kernel.unwrap_or(0);
+            if kernel != 0 {
+                mem_controller.set_kmem_limit(kernel);
             }
 
-            info!(sl!(), "{}", err.desc());
+            let kernel_tcp = memory.kernel_tcp.unwrap_or(0);
+            if kernel_tcp != 0 {
+                mem_controller.set_tcp_limit(kernel_tcp);
+            }
 
-            return Err(anyhow!(e));
+            if memory.swapiness.unwrap_or(0) <= 100 {
+                mem_controller.set_swappiness(memory.swapiness.unwrap_or(0) as u64);
+            }
+            if memory.disable_oom_killer.unwrap_or(false) {
+                match mem_controller.disable_oom_killer() {
+                    Ok(_) => {}
+                    Err(err) => {
+                        error!(sl!(), "failed to disable_oom_killer: {:?}", err);
+                    }
+                }
+            }
+            // res.memory.update_values = true;
         }
 
-        Ok(_) => {}
-    }
+        if r.pids.is_some() {
+            info!(sl!(), "cgroup manager set pids !!!");
+            let pid_controller: &PidController = cg.controller_of().unwrap();
+            let pids = r.pids.as_ref().unwrap();
+            let v = if pids.limit > 0 {
+                MaxValue::Value(pids.limit)
+            } else {
+                MaxValue::Max
+            };
+            pid_controller.set_pid_max(v);
+        }
 
-    Ok(())
-}
+        if r.block_io.is_some() {
+            info!(sl!(), "cgroup manager set block io");
+            res.blkio.update_values = true;
+            let blkio = r.block_io.as_ref().unwrap();
 
-impl Subsystem for CpuSet {
-    fn name(&self) -> String {
-        "cpuset".to_string()
-    }
+            if cg.v2() {
+                res.blkio.weight = convert_blk_io_to_v2_value(blkio.weight);
+                res.blkio.leaf_weight = convert_blk_io_to_v2_value(blkio.leaf_weight);
+            } else {
+                res.blkio.weight = blkio.weight;
+                res.blkio.leaf_weight = blkio.leaf_weight;
+            }
 
-    fn set(&self, dir: &str, r: &LinuxResources, update: bool) -> Result<()> {
-        let mut cpus: &str = "";
-        let mut mems: &str = "";
+            let mut vec = vec![];
+            for d in blkio.weight_device.iter() {
+                let mut w = blkio.weight;
+                let mut lw = blkio.leaf_weight;
+                if cg.v2() {
+                    w = convert_blk_io_to_v2_value(blkio.weight);
+                    lw = convert_blk_io_to_v2_value(blkio.leaf_weight);
+                }
+                let dr = BlkIoDeviceResource {
+                    major: d.blk.major as u64,
+                    minor: d.blk.minor as u64,
+                    weight: w,
+                    leaf_weight: lw,
+                };
+                vec.push(dr);
+            }
+            res.blkio.weight_device = vec;
+
+            let mut vec = vec![];
```

Comment:
While I fully appreciate the beauty of Rust allowing you to reuse the name `vec` here, it might be more readable to have a more descriptive name, e.g. `weight_device_vec`, etc.

Reply:
In addition, these blocks are all very similar, and could be factored for clarity.